x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/index.test.tsx

```diff
@@ -49,12 +49,13 @@ jest.mock('../../../containers/detection_engine/rules/api', () => ({
   createPrepackagedRules: jest.fn(),
 }));
 
-jest.mock('../../../../common/lib/kibana', () => {
-  return {
-    useToast: jest.fn(),
-    useHttp: jest.fn(),
-  };
-});
+// jest.mock('../../../../common/lib/kibana', () => {
+//   return {
+//     useToast: jest.fn(),
+//     useHttp: jest.fn(),
+//     useKibana: jest.fn(),
+//   };
+// });
```

Comment:
TODO: Remove this before merging.

Leaving for now, because that is also something I'm concerned a bit and spent a lot of time debugging and figuring out why this test got broken after I changed the implementation of the callouts.

The reason why the test broke was because the callout components started using local storage via `useKibana`, approximately like that:
callouts -> `useCallOutStorage` -> `useMessagesStorage` -> `useKibana`

The test didn't need this mock, but the mock made it more brittle.

This tests still contains quite a lot of mocks of static dependencies of the page. What are your thoughts guys?

Reply:
The mocks in `common/lib/kibana` are really just meant to be mock objects to satisfy a test interface; IIRC some of them have actual implementations due to a large number of tests relying on them (UI settings tests, I believe?).

From that perspective, the failure here makes sense: a dependency was added but not mocked. The large number of mocks here is indeed a smell; the behavior being tested feels more appropriate for a higher-level test.

I didn't see a definitive question and I'm starting to ramble, but I'll summarize my thoughts:

* Absolutely delete unnecessary mocks if you find them
* `jest.mock` is a fine tool, but I definitely prefer dependency injection wherever possible
* In general: leave things better than you found them (which you are 😉 )
  * inconsistencies and messiness are inevitable in a large codebase like kibana, we've just gotta do our best and remain vigilant 🖖

Reply:
> The mocks in common/lib/kibana are really just meant to be mock objects to satisfy a test interface; IIRC some of them have actual implementations due to a large number of tests relying on them (UI settings tests, I believe?).

Not sure, maybe I misunderstood your point.. but `jest.mock('../../../../common/lib/kibana'` mocked actual code, some very common (as far as I can see) dependencies. I know that books say unit testing = mocking dependencies :) But IMHO whatever makes tests brittle should be avoided. So for this kind of page-level tests I'd probably avoid mocking most of dependencies except smth like local storage etc.

Let me address some individual comments:

> the behavior being tested feels more appropriate for a higher-level test

Oh yeah, 💯 agree. The page is large and we can change it relatively often. Some Cypress tests might be a better option here.

> Absolutely delete unnecessary mocks if you find them

👍 

> jest.mock is a fine tool, but I definitely prefer dependency injection wherever possible

Super agree again.
As a rule of thumb, I'd avoid mocking static deps (imports) if possible. They are more like implicit internal implementation details rather than explicit dependencies.

> In general: leave things better than you found them

I have to admit I agree with that as well!

Thanks @rylnd !